<launch>
    <!-- Configuration file argument -->
    <arg name="config" default="default.yaml" />
    <arg name="rviz" default="false" />

    <node name="height_mapping" pkg="height_mapping_ros" type="height_mapping_node" output="screen">
        <rosparam file="$(find height_mapping_ros)/config/$(arg config)" command="load" subst_value="true" />
        <!-- Override parameters -->
        <param name="config_path" value="$(find height_mapping)/config/global_mapping.yaml" />
        <param name="topics/publish_rate" value="1.0" />
    </node>

    <!-- Launch Rviz with height mapping configuration -->
    <node pkg="rviz" type="rviz" name="$(anon height_mapping_rviz)" if="$(arg rviz)"
          args="-d $(find height_mapping_ros)/launch/rviz/height_mapping.rviz" />

    <!-- Launch Grid Map Visualization -->
    <include file="$(find height_mapping_ros)/launch/grid_map_visualization.launch.xml" />

</launch>